File: Prototype/ShopExample/PrototypeShop/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PrototypeShop
{
    public class Product
    {
        public string Name { get; set; }
        public string Color { get; set; }
        public decimal Price { get; set; }
        public int Size { get; set; }

        public Product(string name, string color, decimal price, int size)
        {
            Name = name;
            Color = color;
            Price = price;
            Size = size;
        }

        public Product Clone()
        {
            return new Product(Name, Color, Price, Size);
        }

        public void ShowInfo()
        {
            Console.WriteLine($"Name: {Name}, Color: {Color}, Price: {Price}, Size: {Size}");
        }
    }
}

File: Prototype/Java/Shop/src/Product.java
import javax.lang.model.element.Name;

public class Product {
    public String name;
    public String color;
    public Double price;
    public Integer size;

    public Product(String name, String color, Double price, Integer size)
    {
        this.name = name;
        this.color = color;
        this.price = price;
        this.size = size;
    }

    public Product Clone()
    {
         return new Product(name, color, price, size);
    }

    public void printInfo()
    {
        System.out.println("Name: " + name);// + "Color: " + Color + "Size: " + Size + "Price: " + price);
        System.out.println("Color: " + color);
        System.out.println("Size: " + size);
        System.out.println("Price: " + price);

    }
}
